src/models/data.py

```diff
@@ -169,43 +226,109 @@ def ds_folder_to_np(self,
+                current=current,
+                historical=historical,
+                pred_months=x_months  # @GABI do we want to include this ?
+            )
+        else:
+            train_data = TrainData(
+                current=None,
+                historical=x_np,
+                pred_months=x_months
+            )
+
+        if self.historical_normalizing_array is not None:
+            if self.experiment == 'nowcast':
+                # normalize the historical X data (all vars including target variable)
+                train_data.historical = (
+                    train_data.historical - self.historical_normalizing_array['mean']
+                ) / (self.historical_normalizing_array['std'])
+
+                # normalize the current X data (no target variable)
+                train_data.current = (
+                    train_data.current - self.current_normalizing_array['mean']
+                ) / (self.current_normalizing_array['std'])
+            else:
+                # only historical variables for non-nowcast experiments
+                train_data.historical = (
+                    train_data.historical - self.historical_normalizing_array['mean']
+                ) / (self.historical_normalizing_array['std'])
 
         assert y_np.shape[0] == x_np.shape[0], f'x and y data have a different ' \
             f'number of instances! x: {x_np.shape[0]}, y: {y_np.shape[0]}'
 
         if clear_nans:
             # remove nans if they are in the x or y data
-            x_nans, y_nans = np.isnan(x_np), np.isnan(y_np)
+            historical_nans, y_nans = np.isnan(train_data.historical), np.isnan(y_np)
 
-            x_nans_summed = x_nans.reshape(x_nans.shape[0],
-                                           x_nans.shape[1] * x_nans.shape[2]).sum(axis=-1)
+            historical_nans_summed = historical_nans.reshape(
+                historical_nans.shape[0], historical_nans.shape[1] * historical_nans.shape[2]
+            ).sum(axis=-1)
             y_nans_summed = y_nans.sum(axis=-1)
 
-            notnan_indices = np.where((x_nans_summed == 0) & (y_nans_summed == 0))[0]
-            x_np, y_np = x_np[notnan_indices], y_np[notnan_indices]
-            x_months = x_months[notnan_indices]
+            notnan_indices = np.where((historical_nans_summed == 0) & (y_nans_summed == 0))[0]
+
+            if self.experiment == 'nowcast':
+                current_nans = np.isnan(train_data.current)
+                current_nans_summed = current_nans.reshape(
+                    current_nans.shape[0], current_nans.shape[1] * current_nans.shape[2]
+                ).sum(axis=-1)
+                notnan_indices = np.where(
+                    (
+                        historical_nans_summed == 0
+                    ) & (y_nans_summed == 0) & (
+                        current_nans_summed == 0
+                    )
+                )[0]
+                train_data.current = train_data.current[notnan_indices]
```

Comment:
The mypy error here is because mypy thinks that `train_data.current` could still be None.

Casting could fix this (or, if thats a nuisance, you can ignore this line by adding the inline comment `type: ignore`)

Reply:
do you mind if i `  # type: ignore`

Reply:
yep! that works :)